needle.pm

```diff
@@ -276,13 +276,34 @@ sub has_tag {
 }
 
 sub has_property {
-    my ($self, $tag) = @_;
-    for my $t (@{$self->{properties}}) {
-        return 1 if ($t eq $tag);
+    my ($self, $property_name) = @_;
+    for my $property (@{$self->{properties}}) {
+        if (ref($property) eq "HASH") {
+            return 1 if ($property->{name} eq $property_name);
+        }
+        else {
+            return 1 if ($property eq $property_name);
+        }
     }
     return 0;
 }
 
+sub get_property_value {
+    my ($self, $property_name) = @_;
+
+    for my $property (@{$self->{properties}}) {
+        if (ref($property) eq "HASH") {
+            return $property->{value} if ($property->{name} eq $property_name);
+        }
+    }
```

Comment:
see above, same as there :)

Reply:
I would not like to change those code to a `grep`. I think this logic of this piece of code is a little complicated than above, so I prefer to keep its readability.